### ISSUE TITLE
单张图inference看效果

### ISSUE THREAD

**Issue by northeastsquare** [valencebond/Strong_Baseline_of_Pedestrian_Attribute_Recognition#13]
attribute是如下列表：   
 attr_list = np.array(['Hat', 'Glasses', 'ShortSleeve', 'LongSleeve', 'UpperStride', 'UpperLogo' \
        , 'UpperPlaid', 'UpperSplice', 'LowerStripe', 'LowerPattern', 'LongCoat', 'Trousers', 'Shorts' \
        , 'Skirt&Dress', 'boots', 'HandBag', 'ShoulderBag', 'Backpack', 'HoldObjectsInFront', 'AgeOver60' \
        , 'Age18-60', 'AgeLess18', 'Female', 'Front', 'Side', 'Back'])

**Comment by northeastsquare** [comment]
运行脚本例子如下
import os, cv2
import pickle
import pprint
from collections import OrderedDict, defaultdict

import numpy as np
import torch
from torch.optim.lr_scheduler import ReduceLROnPlateau
from torch.utils.data import DataLoader
from tqdm import tqdm

from batch_engine import valid_trainer, batch_trainer
from config import argument_parser
from dataset.AttrDataset import AttrDataset, get_transform
from loss.CE_loss import CEL_Sigmoid
from models.base_block import FeatClassifier, BaseClassifier
from models.resnet import resnet50, resnet101
#from models.resnest.resnest import resnest50

from tools.function import get_model_log_path, get_pedestrian_metrics
from tools.utils import time_str, save_ckpt, ReDirectSTD, set_seed, AverageMeter
from PIL import Image

set_seed(605)


def main(args):
    visenv_name = args.dataset
    exp_dir = os.path.join('exp_result', args.dataset)
    model_dir, log_dir = get_model_log_path(exp_dir, visenv_name)
    stdout_file = os.path.join(log_dir, f'stdout_{time_str()}.txt')

    if args.redirector:
        print('redirector stdout')
        ReDirectSTD(stdout_file, 'stdout', False)

    pprint.pprint(OrderedDict(args.__dict__))

    print('-' * 60)
    print(f'use GPU{args.device} for training')
    print(f'train set: {args.dataset} {args.train_split}, test set: {args.valid_split}')

    train_tsfm, valid_tsfm = get_transform(args)
    print(train_tsfm)

    backbone = resnet50()
    classifier = BaseClassifier(nattr=26)
    model = FeatClassifier(backbone, classifier)

    # if torch.cuda.is_available():
    #     model = torch.nn.DataParallel(model).cuda()
    new_state_dict = OrderedDict()
    

    exp_dir = os.path.join('exp_result', args.dataset)
    model_path = os.path.join(exp_dir, args.dataset, 'img_model/pa100k_ckpt_max.pth')
    #model = get_reload_weight(model_path, model)
    checkpoint = torch.load(model_path)
    print("checkpoints:", checkpoint.keys())
    print("checkpoints:", checkpoint.keys())
    state_dict = checkpoint['state_dicts']
    for k, v in state_dict.items():
        name = k[7:] # remove module.
        new_state_dict[name] = v
    print("reloading pretrained models")
    model.load_state_dict(new_state_dict, strict=False)
    model.cuda()
    model.eval()
    preds_probs = []
    gt_list = []
    label_dict = {}
    attr_list = np.array(['Hat', 'Glasses', 'ShortSleeve', 'LongSleeve', 'UpperStride', 'UpperLogo' \
        , 'UpperPlaid', 'UpperSplice', 'LowerStripe', 'LowerPattern', 'LongCoat', 'Trousers', 'Shorts' \
        , 'Skirt&Dress', 'boots', 'HandBag', 'ShoulderBag', 'Backpack', 'HoldObjectsInFront', 'AgeOver60' \
        , 'Age18-60', 'AgeLess18', 'Female', 'Front', 'Side', 'Back'])
    with torch.no_grad():
        #for step, (imgs, gt_label, imgname) in enumerate(tqdm(valid_loader)):
        for root, dirs, files in os.walk('/media/silva/ACB4B3C1B4B38C78/Market-1501-v15.09.15/bounding_box_test'):
            for fl in files:
                fn, ext = os.path.splitext(fl)
                if ext != '.jpg':
                    continue
                impath = os.path.join(root, fl)
                img = Image.open(impath)
                img = img.resize((192, 256))
                orig_img = np.array(img).copy()
                img = valid_tsfm(img)
                imgs = img[None, ...]
                print("imgs.shape", imgs.shape)
                imgs = imgs.cuda()
                valid_logits = model(imgs)
                #print("valid_logits:", valid_logits)
                logits = valid_logits.flatten()
                print("logits:", logits)
                logits = logits.cpu().numpy()
                idx = np.where(logits>0)
                print("idx:", idx)
                print("labels:", attr_list[idx])
                cv2.imshow('img', orig_img[:,:,::-1])
                cv2.waitKey(0)


if __name__ == '__main__':
    parser = argument_parser()
    args = parser.parse_args()

    main(args)

"""
载入的时候要：
from tools.function import LogVisual
sys.modules['LogVisual'] = LogVisual
log = torch.load('./save/2018-10-29_21:17:34trlog')
"""
